base/compiler/abstractinterpretation.jl

```diff
@@ -1290,7 +1295,7 @@ function const_prop_call(interp::AbstractInterpreter,
         return nothing
     end
     @assert inf_result.result !== nothing
-    if concrete_eval_result !== nothing
+    if concrete_eval_result !== nothing && isa(forwarded_argtypes, SimpleArgtypes)
```

Comment:
```suggestion
    if concrete_eval_result !== nothing && !isa(forwarded_argtypes, WidenedSimpleArgtypes)
```

Reply:
I chose to go with the opposite condition, as I think it is more defensive to not overwrite with concrete eval results in case somebody adds more of these.

Reply:
That sounds better.